narwhals/dataframe.py

```diff
@@ -3576,9 +3594,18 @@ def join_asof(
             │ 2019-01-01 00:00:00 ┆ 83.12      ┆ 4696 │
             └─────────────────────┴────────────┴──────┘
         """
-        return super().join_asof(
-            other, left_on=left_on, right_on=right_on, strategy=strategy
-        )
+        if left_on is not None and right_on is not None and on is not None:
+            msg = "Either (`left_on` and `right_on`) or `on` keys should be specified."
+            raise ValueError(msg)
+        if left_on is not None and right_on is not None:
+            return super().join_asof(
+                other, left_on=left_on, right_on=right_on, strategy=strategy
+            )
+        elif on is not None:
+            return super().join_asof(other, on=on, strategy=strategy)
+        else:
+            msg = "Either (`left_on` and `right_on`) or `on` keys should be specified."
+            raise ValueError(msg)
```

Comment:
does this need updating, or can you also just defer to `super().join_asof(` here?

Reply:
thanks, updated this part